connection-cache/src/connection_cache.rs

```diff
@@ -608,14 +614,18 @@ mod tests {
         let port = u16::MAX - MOCK_PORT_OFFSET + 1;
         assert!(port.checked_add(MOCK_PORT_OFFSET).is_none());
         let addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), port);
-        let connection_manager = Box::<MockConnectionManager>::default();
+        let connection_manager = MockConnectionManager::default();
         let connection_cache = ConnectionCache::new(connection_manager, 1).unwrap();
 
         let conn = connection_cache.get_connection(&addr);
         // We (intentionally) don't have an interface that allows us to distinguish between
         // UDP and Quic connections, so check instead that the port is valid (non-zero)
         // and is the same as the input port (falling back on UDP)
-        assert!(conn.server_addr().port() != 0);
-        assert!(conn.server_addr().port() == port);
+        assert_ne!(port, 0u16);
+        assert_eq!(BlockingClientConnection::server_addr(&*conn).port(), port);
+        assert_eq!(
```

Comment:
conn is blocking, I don't think we need check the nonblocking case

Reply:
It implements both interfaces. Does not hurt to assert both.

Reply:
The type it returns is a `MockUdpConnection` which implements `NonblockingClientConnection`.
https://github.com/solana-labs/solana/blob/d53c49c23/connection-cache/src/connection_cache.rs#L525

Again, it is an `assert` inside test. Does not hurt to keep it.

Reply:
That is fine. But it is still confusing -- it does not sound like testing the interface of get_connection -- instead it is testing the test code itself having both implementations.